File: backend/app/models/product.py
from datetime import datetime, timezone
from decimal import Decimal

from sqlalchemy import DateTime, Numeric, String
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base

class Product(Base):
    __tablename__ = "products"

    id: Mapped[int] = mapped_column(primary_key=True)

    itemno: Mapped[str] = mapped_column(String(50), unique=True, index=True, nullable=False)
    buying_price: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    selling_price: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    image_path: Mapped[str | None] = mapped_column(String(500), nullable=True)
    colors: Mapped[str | None] = mapped_column(String(500), nullable=True)
    colli: Mapped[int | None] = mapped_column(nullable=True)
    tags: Mapped[str | None] = mapped_column(String(255), nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc), nullable=False)

File: frontend/src/types/product.ts
export interface Product { id: number; itemno: string; buying_price: string; selling_price: string; image_path: string | null; colors: string | null; colli: number | null; tags: string | null; created_at: string; updated_at: string }
export interface ProductPayload { itemno: string; buying_price: number; selling_price: number; image_path: string | null; colors: string | null; colli: number | null; tags: string | null }
export interface ProductFormValues { itemno: string; buying_price: string; selling_price: string; colli: string; tags: string; productImage: File | null; colorsImage: File | null; image_path: string | null; colors: string | null }
export type SortOption = 'created_desc' | 'created_asc' | 'itemno_asc' | 'itemno_desc'
